.github/workflows/pr.yml

```diff
@@ -0,0 +1,11 @@
+name: Lint Commit Messages
+on: [pull_request]
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
i think you need to apply the fetch depth also to your publish action to not miss out on commits, or do you tend to squash them? (if you can commit to master directly and only push after doing a couple local changes you might have the same issue) 🤔